### DISCUSSION TITLE
How to integrate "MACADDR8 & TIMESTAMPTZ"

### DISCUSSION THREAD

**Discussion by asimfarooq5** [sqlalchemy/sqlalchemy#8388] [Usage Questions]
As postgres supports MACADDR8 & TIMESTAMPTZ, 
how to integrate with sqlalchemy ?

**Comment by zzzeek** [comment]
please see:

https://docs.sqlalchemy.org/en/14/dialects/postgresql.html#sqlalchemy.dialects.postgresql.MACADDR

https://docs.sqlalchemy.org/en/14/dialects/postgresql.html#sqlalchemy.dialects.postgresql.TIMESTAMP

**Reply by zzzeek** [reply]
the latter is TIMESTAMP WITH TIMEZONE, if that's the same thing as TIMESTAMPTZ.

MACADDR8 you can use [UserDefinedType](https://docs.sqlalchemy.org/en/14/core/custom_types.html?highlight=userdefinedtype#sqlalchemy.types.UserDefinedType) or create an issue + pull request that follows our [PR guidelines](https://www.sqlalchemy.org/develop.html#pullrequests)

**Reply by asimfarooq5** [reply]
I have found a workaround for this:

``` python
class MACADDR8(MACADDR):
    __visit_name__ = "MACADDR8"


class TIMESTAMPTZ(TIMESTAMP):
    __visit_name__ = "TIMESTAMPTZ"
```

``` python

ischema_names = {
    "macaddr8": MACADDR8,
    "timestampz": TIMESTAMPZ
}
```

``` python
    def visit_MACADDR8(self, type_, **kw):
        return "MACADDR8"

    def visit_TIMESTAMPTZ(self, type_, **kw):
        return "TIMESTAMPTZ"
```

**Reply by zzzeek** [reply]
> ```python
> 
>     def visit_TIMESTAMPTZ(self, type_, **kw):
>         return "TIMESTAMPTZ"
> ```

this one is unnecessary and might not work correctly in all cases, as you are not setting the timezone=True attribute.   Please use TIMESTAMP(timezone=True).  

https://www.postgresql.org/docs/current/datatype-datetime.html

> timestamptz is accepted as an abbreviation for timestamp with time zone; this is a PostgreSQL extension.
